src/AWSCredentials.jl

```diff
@@ -185,263 +190,311 @@ function localhost_is_ec2()
-                * "... ")
-    end
+# Arguments
+- `profile::AbstractString`: Specific profile used to get AWSCredentials
+- `ini::Inifile`: Inifile to look into for the `profile` credentials
+- `config::Bool`: True if using `~/.aws/config`
 
-    key_id = get(ini, profile, "aws_access_key_id")
-    key = get(ini, profile, "aws_secret_access_key")
+Get `AWSCredentials` for the specified `profile` from the `inifile`. If targeting the `~/.aws/config`
+file, with a non-default `profile`, you must specify `config=true` otherwise the default credentials
+will be returned.
+"""
+function _aws_get_credential_details(profile::AbstractString, ini::Inifile, config::Bool)
+    access_key = get(ini, profile, "aws_access_key_id")
+    secret_key = get(ini, profile, "aws_secret_access_key")
     token = get(ini, profile, "aws_session_token", "")
 
     if config
         profile = "profile $profile"
-        key_id = get(ini, profile, "aws_access_key_id", key_id)
-        key = get(ini, profile, "aws_secret_access_key", key)
+        access_key = get(ini, profile, "aws_access_key_id", access_key)
+        secret_key = get(ini, profile, "aws_secret_access_key", secret_key)
         token = get(ini, profile, "aws_session_token", token)
     end
 
-    (key, key_id, token)
+    return (access_key, secret_key, token)
 end
 
-function aws_get_profile()
-    get(ENV, "AWS_DEFAULT_PROFILE", get(ENV, "AWS_PROFILE", "default"))
-end
 
+"""
+    aws_get_region(profile::AbstractString, ini::Inifile)
+
+# Arguments
+- `profile::AbstractString`: Specific profile used to get the region
+- `ini::Inifile`: Inifile to look in for the region
+
+Retrieve the AWS Region for a given profile, returns `us-east-1` as a default.
+"""
 function aws_get_region(profile::AbstractString, ini::Inifile)
     region = get(ENV, "AWS_DEFAULT_REGION", "us-east-1")
-
     region = get(ini, profile, "region", region)
     region = get(ini, "profile $profile", "region", region)
+
+    return region
 end
 
-function aws_get_role(role::AbstractString, ini::Inifile)
-    source_profile, role_arn = aws_get_role_details(role, ini)
-    source_profile === :notfound && return nothing
+"""
+    aws_user_arn(aws::AWSConfig)
+
+# Arguments
+- `aws::AWSConfig`: SymbolDict used to retrieve the user arn
+
+Retrieve the user ARN from the `AWSConfig`, if not present query Security Token Services (STS).
+"""
+function aws_user_arn(aws::AWSConfig)
+    creds = aws[:creds]
+    if creds.user_arn == ""
+        r = Services.sts(aws, "GetCallerIdentity", [])
+        creds.user_arn = r["Arn"]
+        creds.account_number = r["Account"]
+    end
+    return creds.user_arn
+end
+
+"""
+    aws_account_number(aws::AWSConfig)
 
-    if debug_level > 0
-        println("Assuming \"$source_profile\"... ")
+# Arguments
+- `aws::AWSConfig`: SymbolDict used to retrieve the AWS account number
+
+Retrieve the `AWS account number` from the `AWSConfig` object.
+
+BUG: https://github.com/JuliaCloud/AWSCore.jl/issues/87
```

Comment:
Should be a comment. End-users don't really need to be aware of that implementation detail